docs/en/wallet-solution.rst

```diff
@@ -175,22 +217,155 @@ A Wallet Provider instead is responsible for:
+
+    If any errors occur in any DIS process, such as device integrity verification, for example, due to an unavailable DIS, an internal error, or an invalid nonce in the integrity request, the DIS raises an error response (e.g., see  `Play Integrity API Errors <Play Integrity API Errors_>`_). The Wallet Instance MUST process these errors accordingly. Details on error handling are left to the Wallet Instance implementation.
+ 
+
+**Step 8**: The Device Integrity Service performs the following actions:
+
+* Creates a Key Attestation that is linked with the provided "challenge" and the public key of the Wallet Hardware.
+* Incorporates information pertaining to the device's security.
+* Uses an OEM private key to sign the Key Attestation, therefore verifiable with the related OEM certificate, confirming that the Cryptographic Hardware Keys are securely managed by the operating system.
+
+**Step 9 (Wallet Instance Registration Request)**: The Wallet Instance sends a request to the `Wallet Instance Management endpoint`_ of the Wallet Provider Backend to register the Wallet Instance, identified by the Cryptographic Hardware Key public key. 
+The request body includes the following claims: the ``challenge``, Key Attestation (``key_attestation``), and Cryptographic Hardware Key Tag (``hardware_key_tag``).
+
+Below is a non-normative example of a Wallet Instance Registration Request.
+
+.. code-block:: http
+
+    POST /wallet-instances HTTP/1.1
+    Host: walletprovider.example.com
+    Content-Type: application/json
+
+    {
+      "challenge": "0fe3cbe0-646d-44b5-8808-917dd5391bd9",
+      "key_attestation": "o2NmbXRvYXBwbGUtYXBw... redacted",
+      "hardware_key_tag": "WQhyDymFKsP95iFqpzdEDWW4l7aVna2Fn4JCeWHYtbU="
+    }
+
+.. note::
+  It is not necessary to send the Wallet Hardware public key because it is already included in the ``key_attestation``.
+  As seen in the previous steps, the Device Integrity Service (DIS) creates a Key Attestation linked to the provided "challenge" and the public key of the Wallet Hardware. This process eliminates the need to send the Wallet Hardware public key directly, as it is already included in the key attestation. The ``hardware_key_tag`` serves as a reference or identifier for the corresponding Cryptographic Hardware key stored by the Wallet Provider. Therefore, the Wallet Provider can associate the received ``hardware_key_tag`` with the appropriate Cryptographic Hardware key in its storage.
+
+.. warning::
+  During the registration phase of the Wallet Instance with the Wallet Provider it is also necessary to associate it with a specific user
+  uniquely identifiable by the Wallet Provider. This association is at the discretion of the Wallet Provider and will not be addressed
+  within these guidelines as each Wallet Provider may or may not have a user identification system already implemented.
```

Comment:
```suggestion
  During the registration phase of the Wallet Instance with the Wallet Provider it is also necessary to associate the Wallet Instace with a specific User, authenticating the User with the Wallet Provider. The authentication mechanism is at the discretion of the Wallet Provider and it will not be addressed within these guidelines, as each Wallet Provider may have its User authentication systems already implemented.
```